src/classes/CompensationsController.cls

```diff
@@ -0,0 +1,21 @@
+public with sharing class CompensationsController {
+
+	@AuraEnabled
+	public static List<Compensation__c> getCompensations(String recordType, String contactId) {
+        System.debug(recordType);
+        System.debug(contactId);
+        if (recordType == 'all'){
+            if (contactId != null){
+                return [SELECT Id, Name, Contact__r.Id, Contact__r.Name, Contact__r.Birthdate, Job_Category__c, RecordType.Name, Amount__c, Location__c, Office__c, Submitted__c, Status__c FROM Compensation__c WHERE Contact__r.Id =:contactId];
```

Comment:
all queries should be limited so we prevent our user from hitting any governor limits regarding the result set's size.